From ff4fa6cf60b5b35ffae04a9499b87f540495e32d Mon Sep 17 00:00:00 2001
From: Muthukumar K <muthukumar.kandasamy@intel.com>
Date: Sat, 24 Nov 2012 17:26:05 +0530
Subject: [PORT FROM MAIN] three little Shelf drum on the left and gong on the above cannot sound

BZ: 66069

While finding the duration of ogg files, the last granule
position is searched. While doing this, finding EOF is
considered as an error, whereas it should be considered
as a warning and the granule position should be searched
again by moving the current pointer still back. This fixes
the issue.

Change-Id: I2897526c5fa55f800da802b2a325445c697a42b7
Orig-Change-Id: I4205e48af435feed408a6af7e31f08977c01a5ac
Signed-off-by: Muthukumar K <muthukumar.kandasamy@intel.com>
Reviewed-on: http://android.intel.com:8080/80741
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/libstagefright/OggExtractor.cpp |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/OggExtractor.cpp b/media/libstagefright/OggExtractor.cpp
index 776dfb1..2cafa3a 100644
--- a/media/libstagefright/OggExtractor.cpp
+++ b/media/libstagefright/OggExtractor.cpp
@@ -275,6 +275,11 @@ status_t MyVorbisExtractor::findPrevGranulePosition(
         ALOGV("backing up %lld bytes", pageOffset - prevGuess);
 
         status_t err = findNextPage(prevGuess, &prevPageOffset);
+        // If reaches EOF, try to find again.
+        if (prevGuess != 0 && err == ERROR_END_OF_STREAM) {
+            continue;
+        }
+        // Some unknown error. So break the loop.
         if (err != OK) {
             return err;
         }
-- 
1.7.4.1

